proposals/nnnn-video-streaming-state.md

```diff
@@ -0,0 +1,165 @@
+# Enhancing onHMIStatus with a New Parameter for Video Streaming State
+
+* Proposal: [SDL-NNNN](nnnn-video-streaming-state.md)
+* Author: [Zhimin Yang](https://github.com/smartdevicelink/yang1070)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / iOS / Android / RPC]
+
+## Introduction
+
+This proposal adds a new parameter `videoStreamingState` to `onHMIStatus` notification so that the SDL can notify an app stop video streaming. This helps SDL handle the use cases of the coexistence of a navigation app and a projection app or multiple navigation/projection apps.
+
+## Motivation
+
+Currently, without Projection apps, a mobile navigation app can start video streaming in HMI level `FULL` and can continue streaming while in HMI level `LIMITED` in order to have a good user experience. The app must stop video streaming after receiving `onHMIStatus` with HMI level `NONE` or `BACKGROUND`. If the app does not stop streaming after certain time, SDL will send a `StopService` Control Frame to the app in protocol layer. Therefore, the HMI level implies when a navigation app shall stop video streaming or not. There is no dedicate RPC message or parameter from SDL to mobile app to say ‘stop video streaming’ in application layer. 
+
+With the new projection apps coming, HMI level itself is not enough to tell when a navigation app or a projection app shall stop video streaming. For example, a driver launches a navigation app first, launches a non-media projection app next. The driver shall still be able to hear turn-by-turn instructions. The navigation app is in HMI level `LIMITED` with `audioStreamingState`=`AUDIBLE`. Because the projection app is streaming, the navigation app shall stop video streaming while it is still in `LIMITED`. (Compare to the case that the driver launches a navigation app and a media app in sequence. The navigation app is in HMI level `LIMITED` with `audioStreamingState`=` ATTENUATED`. It continues streaming video.)
+
+## Proposed solution
+
+We propose to add a new parameter `videoStreamingState` to `onHMIStatus` to explicitly notify the app that it must stop video streaming.
+SDL maintains which app can do video streaming or equivalently we say which app is a video source. If the driver launches a SDL app, SDL knows whether the app uses video service or not depending on the `AppHMIType`. Currently only `NAVIGATION` and `PROJECTION` app use video service. Other types cannot not use video service. Note, an app can have multiple `AppHMIType`s.
+
+Assume there is at most one app that can do video streaming at any time, the following table shows how SDL shall do in different cases.
+
+
+| Case# | Current Video source | SDL App2 comes to FULL | Results | SDL Actions regarding videoStreamingState | 
```

Comment:
I believe "SDL Actions" should be "SDL Core Actions"

Reply:
updated